### PR TITLE
feat: plugin unit tests

### LINKED ISSUE #24: Add unit tests for plugin code
The Keycloak Java plugin was a port from existing behavior in https://repo1.dso.mil/big-bang/product/plugins/keycloak-p1-auth-plugin

Wanted to track the need to pull in (and or maybe add/update?) unit tests from https://repo1.dso.mil/big-bang/product/plugins/keycloak-p1-auth-plugin/-/tree/main/p1-keycloak-plugin/src/test

cc @jeff-mccoy

### PR DESCRIPTION
**Pull request by UnicornChance**
## Description
* Add P1 Plugin Unit Tests 
* Configure CI job for testing
* Add docs for developing and viewing reports 
* Task for local dev

### Question regarding fips and the future of keycloak, do we want to include these dependencies as well?
* keycloak-crypto-fips1402 : cryptography algorithms for fips 
* bc-fips : bouncy castle general fips library that is supposed to be used in conjunction with bcpg-fips ( the dep we are already using)
```
        <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-crypto-fips1402</artifactId>
            <version>${keycloak.version}</version>
        </dependency>
        <dependency>
            <groupId>org.bouncycastle</groupId>
            <artifactId>bc-fips</artifactId>
            <version>1.0.2.4</version>
        </dependency> 
```

Tested by creating image and deploying with uds-core ( @rjferguson21 is going to test CAC integration to make sure nothing broke ). 

Tested CI in local repo. Maven-Surefire reports say 100% passing unit tests and ~80% code coverage. 

Can test unit tests with `uds run dev-plugin` from base directory of `uds-identity-config`. 

## Related Issue

Fixes #24

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: src/plugin/pom.xml [hunk 1]
```diff
@@ -36,11 +36,48 @@
...
+            <artifactId>keycloak-crypto-fips1402</artifactId>
+            <version>${keycloak.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bc-fips</artifactId>
+            <version>1.0.2.4</version>
+        </dependency> 
+        -->
+
```
Comment (rjferguson21) [inline src/plugin/pom.xml:58]:
```suggestion
```
Can we get rid of this block and maybe post the same question in the PR?

**Comment by rjferguson21** [comment]
Tested this locally and CAC auth is working.

File: src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/X509Tools.java [hunk 2]
```diff
@@ -209,4 +209,4 @@ private static Object getX509Identity(final KeycloakSession session, final HttpR
         return null;
     }
 
-}
+}
```
Comment (rjferguson21) [inline src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/X509Tools.java:212]: nit: Can we make sure to leave the new lines at the end of these files?

**Review by rjferguson21 — APPROVED** [review]

**Comment by UnicornChance** [comment]
@jeff-mccoy @mjnagel @rjferguson21   Wanted to tag you all and see if you have any input on the mentioned bouncy-castle fips libraries. Registration bugfix in here so want to get this merged. Can move this conversation to an issue for FIPS specifically.

**Review by rjferguson21 — APPROVED** [review]

**Pull request merged**